core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -360,6 +360,24 @@ sealed abstract class Parser0[+A] { self: Product =>
   def withContext(str: String): Parser0[A] =
     Parser.withContext0(this, str)
 
+  /** return the result of the parser together
+    * with the position of the starting of the consumption
+    */
+  def withCaret: Parser0[(A, Caret)] =
```

Comment:
what about call this `def caretBefore: Parser0[(Caret, A)]` or something?

I'm really nervous `withCaret` and the fact that the caret is on the right is going to suggest to people that the caret is after the parse.